openwisp_network_topology/admin.py

```diff
@@ -1,12 +1,15 @@
 import swapper
+from django import forms
 from django.conf.urls import url
 from django.contrib import admin, messages
 from django.contrib.admin import ModelAdmin
 from django.db.models import Q
 from django.template.response import TemplateResponse
 from django.templatetags.static import static
 from django.urls import reverse
+from django.utils.safestring import mark_safe
 from django.utils.translation import ugettext_lazy as _
+from flat_json_widget.widgets import FlatJsonWidget
```

Comment:
this module needs to be added here: https://github.com/openwisp/openwisp-network-topology/blob/master/requirements.txt

openwisp-controller is not a direct dependency. The build is not failing because openwisp-controller is being installed with an additional command to test the integration module.

Reply:
Thank you for this catch. I forgot to add it :smile: . But the build isn't failing because of this. the Build is failing because [modify_settings](https://github.com/openwisp/openwisp-network-topology/blob/master/openwisp_network_topology/integrations/device/tests.py#L265) is called. I tried some many fix but they didn't work. I don't think the fix should be done in this module. if we fix it here, we might be hiding a potential bug which will certainly surface again one day.